Bottom: e394e37491899340294bfd977f43c098c558c2b1
Top:    e7a76e5bc200b56429777851aab7bce4bcbb7c56
Author: Alexandre Oliva <aoliva@redhat.com>
Date:   2013-08-13 15:29:40 -0300

Thread safety documentation.

for  ChangeLog

	* manual/locale.texi: Document thread safety properties.


---

diff --git a/manual/locale.texi b/manual/locale.texi
index 2f10fcd..323268d 100644
--- a/manual/locale.texi
+++ b/manual/locale.texi
@@ -224,6 +224,10 @@ The symbols in this section are defined in the header file @file{locale.h}.
 @comment locale.h
 @comment ISO
 @deftypefun {char *} setlocale (int @var{category}, const char *@var{locale})
+@safety{@mtunsafe{uunguard}}
+@c This function is MT-Safe, but uses of the global locale object are
+@c unguarded in functions that ought to be MT-Safe, so we're ruling out
+@c the use of this function once threads are started.
 The function @code{setlocale} sets the current locale for category
 @var{category} to @var{locale}.  A list of all the locales the system
 provides can be created by running
